tasmota/tasmota_xsns_sensor/xsns_96_flowratemeter.ino

```diff
@@ -25,34 +25,35 @@
 #define XSNS_96                       96
 
 
-
 #define FLOWRATEMETER_WEIGHT_AVG_SAMPLE   20  // number of samples for smooth weigted average
 #define FLOWRATEMETER_MIN_FREQ             1  // Hz
 
-#define D_JSON_FLOWRATEMETER_RATE         "Rate"
-#define D_JSON_FLOWRATEMETER_VALUE        "Value"
-#define D_JSON_FLOWRATEMETER_UNIT         "Unit"
-#define D_JSON_FLOWRATEMETER_VALUE_AVG    "average"
-#define D_JSON_FLOWRATEMETER_VALUE_RAW    "raw"
-
-
+#define D_JSON_FLOWRATEMETER_RATE           "Rate"
+#define D_JSON_FLOWRATEMETER_VALUE          "Source"
+#define D_JSON_FLOWRATEMETER_UNIT           "Unit"
+#define D_JSON_FLOWRATEMETER_AMOUNT_TODAY   "AmountToday"
+#define D_JSON_FLOWRATEMETER_AMOUNT_UNIT    "AmountUnit"
+#define D_JSON_FLOWRATEMETER_DURATION_TODAY "DurationToday"
+#define D_JSON_FLOWRATEMETER_VALUE_AVG      "average"
+#define D_JSON_FLOWRATEMETER_VALUE_RAW      "raw"
 
-#ifdef USE_WEBSERVER
-const char HTTP_SNS_FLOWRATEMETER[] PROGMEM =
-   "{s}" D_FLOWRATEMETER_NAME "-%d{m}%*_f %s{e}"
-   ;
-#endif  // USE_WEBSERVER
 
+#define FLOWRATEMETER_INVALID (uint32_t)-1
 
 int32_t flowratemeter_period[MAX_FLOWRATEMETER] = {0};
 float   flowratemeter_period_avg[MAX_FLOWRATEMETER] = {0};
 uint32_t flowratemeter_count[MAX_FLOWRATEMETER] = {0};
-volatile uint32_t flowratemeter_last_irq[MAX_FLOWRATEMETER] = {0};
+volatile uint32_t flowratemeter_last_irq[MAX_FLOWRATEMETER] = {FLOWRATEMETER_INVALID};
```

Comment:
Although it was already there, I don't think that `volatile` does anything here. It's neither a mapped address, nor have we multi-threading in Tasmota.

Reply:
these vars are use by IRQ function and outside together (like multi-threading)

Reply:
Ah yes! Of course. I forgot about IRQ drivers.